controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -88,6 +88,31 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 			response[name] = errors.Wrap(err, "failed to list etcd members using etcd client")
 			continue
 		}
+
+		// Check if any member's node is missing from workload cluster
+		// If any, delete it with best effort
+		for _, m := range members {
+			nodeName := m.Name
+			isFound := false
+			for _, node := range controlPlaneNodes.Items {
+
+				name := node.Name
+				if nodeName == name {
+					isFound = true
```

Comment:
You can probably remove `isFound` and just `continue` here given that we found the node in the node list